modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/portlet/action/EditAccountEntryContactMVCActionCommand.java

```diff
@@ -0,0 +1,154 @@
+ */
+
+package com.liferay.account.admin.web.internal.portlet.action;
+
+import com.liferay.account.constants.AccountActionKeys;
+import com.liferay.account.constants.AccountPortletKeys;
+import com.liferay.account.model.AccountEntry;
+import com.liferay.account.service.AccountEntryService;
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.Contact;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.service.ContactService;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Danny Situ
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + AccountPortletKeys.ACCOUNT_ENTRIES_ADMIN,
+		"javax.portlet.name=" + AccountPortletKeys.ACCOUNT_ENTRIES_MANAGEMENT,
+		"mvc.command.name=/account_admin/edit_account_entry_contact"
+	},
+	service = MVCActionCommand.class
+)
+public class EditAccountEntryContactMVCActionCommand
+	extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		_checkPermission(actionRequest);
+
+		String cmd = ParamUtil.getString(actionRequest, Constants.CMD);
+
+		try {
+			if (cmd.equals(Constants.ADD)) {
+				_addAccountEntryContact(actionRequest);
+			}
+			else if (cmd.equals(Constants.UPDATE)) {
+				_updateAccountEntryContact(actionRequest);
+			}
+
+			String redirect = ParamUtil.getString(actionRequest, "redirect");
+
+			if (Validator.isNotNull(redirect)) {
+				sendRedirect(actionRequest, actionResponse, redirect);
+			}
+		}
+		catch (Exception exception) {
+			if ((exception instanceof ModelListenerException) &&
+				(exception.getCause() instanceof PortalException)) {
+
+				throw (PortalException)exception.getCause();
+			}
+
+			throw exception;
+		}
+	}
+
+	private void _addAccountEntryContact(ActionRequest actionRequest)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		long accountEntryId = ParamUtil.getLong(
+			actionRequest, "accountEntryId");
+
+		AccountEntry accountEntry = _accountEntryService.getAccountEntry(
+			accountEntryId);
```

Comment:
Do we need this to performe some permission check or we can use directly the accountEntryId in the next method call?

Reply:
oh, yeah I originally added it in as a check to see if the accountEntryId is a valid one, but looks like its getting checked at _checkPermission method. We can just pass the accountEntryId directly.